chapter3/src/3_2/3_2_2/main_3_2_2_3.rs

```diff
@@ -0,0 +1,40 @@
+/**
+ *
+ * chapter 3_2_2_3
+ * copy with buffer
+ *
+ */
+use std::io::{Read, Result, Write};
+
+pub fn main() -> std::io::Result<()> {
+    let mut read: &[u8] = &b"abcdefg".repeat(500 * 1024); //358.4KB Source
+    let mut dst: Vec<u8> = vec![];
+    let mut buffer: Box<[u8]> = Box::new([0; 8 * 1024]); //8KB Buffer
```

Comment:
ここは `Box` にしなくても解決できると思います！コンパイルタイムでバッファサイズを計算済みにして配列を確保してもらえばよいので、サイズ部分に計算済みの値である `8192` を直接入れるか、もしくは書籍の書き方を尊重するならば `const` で用意する手が取れるかなと思います。

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=57e4d33343f20d83f1064c8507d5ae32

```suggestion
    const BUFFER_SIZE: usize = 8 * 1024;
    let mut buffer: [u8; BUFFER_SIZE] = [0; BUFFER_SIZE]; //8KB Buffer
```

Reply:
頂いたコードのほうが良さそうなのでsugestionでcommitしました